module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include <algorithm>
+#include <iostream>
+
+#include "compression.hpp"
+
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& inputMap){
+    std::vector<std::pair<uint8_t, uint8_t>> outputVec {};
+    uint8_t color = 0; 
+    int value = 0;
+
+    for(const auto& mapRow : inputMap){
+        color = mapRow[0]; 
+        value = 0;
+
+        for(auto rowElement : mapRow){
```

Comment:
These element are not const? :)

Reply:
I agree with @kuchichan  better add here const ;)